services/web/server/src/simcore_service_webserver/login/_registration.py

```diff
@@ -74,47 +87,78 @@ def ensure_enum(cls, v):
 }
 
 
+async def _raise_if_registered_in_product(app: web.Application, user_email, product):
+    # NOTE on `product.group_id is None`: A user can be registered in one of more products only if product groups are defined
+    if product.group_id is None or await is_user_by_email_in_group(
+        app, user_email=user_email, group_id=product.group_id
+    ):
+        raise web.HTTPConflict(
+            reason=MSG_EMAIL_ALREADY_REGISTERED,
+            content_type=MIMETYPE_APPLICATION_JSON,
+        )
+
+
 async def check_other_registrations(
+    app: web.Application,
     email: str,
+    current_product: Product,
     db: AsyncpgStorage,
     cfg: LoginOptions,
 ) -> None:
+    # An account is already registered with this email
     if user := await db.get_user({"email": email}):
-        # An account already registered with this email
-        #
-        #  RULE 'drop_previous_registration': any unconfirmed account w/o confirmation or
-        #  w/ an expired confirmation will get deleted and its account (i.e. email)
-        #  can be overtaken by this new registration
-        #
-        if user["status"] == CONFIRMATION_PENDING:
-            _confirmation = await db.get_confirmation(
-                filter_dict={
-                    "user": user,
-                    "action": ConfirmationAction.REGISTRATION.value,
-                }
-            )
-            drop_previous_registration = not _confirmation or is_confirmation_expired(
-                cfg, _confirmation
-            )
-            if drop_previous_registration:
-                if not _confirmation:
-                    await db.delete_user(user=user)
-                else:
-                    await db.delete_confirmation_and_user(
-                        user=user, confirmation=_confirmation
-                    )
+        user_status = UserStatus(user["status"])
+        match user_status:
 
-                _logger.warning(
-                    "Re-registration of %s with expired %s"
-                    "Deleting user and proceeding to a new registration",
-                    f"{user=}",
-                    f"{_confirmation=}",
+            case UserStatus.ACTIVE:
+                await _raise_if_registered_in_product(
+                    app, user_email=user["email"], product=current_product
                 )
-                return
 
-        raise web.HTTPConflict(
-            reason=MSG_EMAIL_EXISTS, content_type=MIMETYPE_APPLICATION_JSON
-        )
+            case UserStatus.CONFIRMATION_PENDING:
+                # An account already registered with this email
+                #
+                #  RULE 'drop_previous_registration': any unconfirmed account w/o confirmation or
+                #  w/ an expired confirmation will get deleted and its account (i.e. email)
+                #  can be overtaken by this new registration
+                #
+                # resend confirmation ??
```

Comment:
Did you forget this one? Do you still need to do it?

Reply:
I forgot to remove the comment. When the notifier service is in place, I will be resending some emails in situations like this. for the moment i am avoiding it so i do not collapse more the webserver